rust/agama-server/src/users/web.rs

```diff
@@ -0,0 +1,253 @@
+    state.users.remove_first_user().await?;
+    Ok(())
+}
+
+#[utoipa::path(put, path = "/users/user", responses(
+    (status = 200, description = "User values are set"),
+    (status = 400, description = "The D-Bus service could not perform the action"),
+))]
+async fn set_first_user(
+    State(state): State<UsersState<'_>>,
+    Json(config): Json<FirstUser>,
+) -> Result<(), Error> {
+    state.users.set_first_user(&config).await?;
+    Ok(())
+}
+
+#[derive(Clone, Debug, Default, Serialize, Deserialize, utoipa::ToSchema)]
+pub struct RootPasswordSettings {
+    pub value: String,
+    pub encrypted: bool,
+}
+
+#[utoipa::path(delete, path = "/users/root_password", responses(
+    (status = 200, description = "Removes the root password"),
+    (status = 400, description = "The D-Bus service could not perform the action"),
+))]
+async fn remove_root_password(State(state): State<UsersState<'_>>) -> Result<(), Error> {
+    state.users.remove_root_password().await?;
+    Ok(())
+}
+
+#[utoipa::path(put, path = "/users/root_password", responses(
+    (status = 200, description = "Root password is set"),
+    (status = 400, description = "The D-Bus service could not perform the action"),
+))]
+async fn set_root_password(
+    State(state): State<UsersState<'_>>,
+    Json(config): Json<RootPasswordSettings>,
+) -> Result<(), Error> {
+    state
+        .users
+        .set_root_password(&config.value, config.encrypted)
+        .await?;
+    Ok(())
+}
+
+#[utoipa::path(delete, path = "/users/root_sshkey", responses(
+    (status = 200, description = "Removes the root SSH key"),
+    (status = 400, description = "The D-Bus service could not perform the action"),
+))]
+async fn remove_root_sshkey(State(state): State<UsersState<'_>>) -> Result<(), Error> {
+    state.users.set_root_sshkey("").await?;
+    Ok(())
+}
+
+#[utoipa::path(put, path = "/users/root_sshkey", responses(
+    (status = 200, description = "Root SSH Key is set"),
+    (status = 400, description = "The D-Bus service could not perform the action"),
+))]
+async fn set_root_sshkey(
+    State(state): State<UsersState<'_>>,
+    Json(key): Json<String>,
+) -> Result<(), Error> {
+    state.users.set_root_sshkey(key.as_str()).await?;
+    Ok(())
+}
+
+#[derive(Clone, Debug, Default, Serialize, Deserialize, utoipa::ToSchema)]
+pub struct RootInfo {
+    password: bool,
+    sshkey: Option<String>,
+}
+
+#[derive(Clone, Debug, Default, Serialize, Deserialize, utoipa::ToSchema)]
+pub struct UsersInfo {
+    user: Option<FirstUser>,
+    root: RootInfo,
+}
+
+#[utoipa::path(put, path = "/users/config", responses(
+    (status = 200, description = "Configuration for users including root and the first user", body = UsersInfo),
+    (status = 400, description = "The D-Bus service could not perform the action"),
+))]
+async fn get_info(State(state): State<UsersState<'_>>) -> Result<Json<UsersInfo>, Error> {
```

Comment:
Here is where the API looks confusing to me. If I need to update the user, I do call `PUT` on the `/users/user` route. But, if I want to get the current user, I need to get the info from `/users/config`.

Reply:
yeap, that is why I mention that we can have that `/users/config` and patch when need to change it. Question is how much it will be consistent with other APIs we have.